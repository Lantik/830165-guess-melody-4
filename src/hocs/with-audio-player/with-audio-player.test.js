import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import withAudioPlayer from './with-audio-player.js';

const mock = {
  song: {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    id: 0
  }
};

const TestComponent = ({renderPlayer, song}) => {
  const {src, id} = song;

  return (
    <div>{renderPlayer(src, id)}</div>
  );
};

TestComponent.propTypes = {
  renderPlayer: PropTypes.func.isRequired,
  song: PropTypes.shape({
    src: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired
  }).isRequired
};

it(`Hoc adds audio player to component`, () => {
  const {song} = mock;
  const TestComponentWithAudioPlayer = withAudioPlayer(TestComponent);

  let tree = renderer
    .create(<TestComponentWithAudioPlayer
      song={song}
    />, {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

  expect(tree).toMatchSnapshot();
});
