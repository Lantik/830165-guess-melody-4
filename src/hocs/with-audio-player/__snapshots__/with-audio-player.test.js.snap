// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hoc adds audio player to component 1`] = `
<div>
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
</div>
`;
